### PR TITLE
Add an optional block to the PyLint regex which extracts the priority, if possible

### PR DESCRIPTION
**Pull request by AngledLuffa**

### CODE REVIEW

File: src/main/java/hudson/plugins/warnings/parser/PyLintParser.java [hunk 1]
```diff
@@ -34,12 +43,22 @@ protected boolean isLineInteresting(final String line) {
...
-        //First letter of the Pylint classification is one of F/E/W/R/C. E/F/W are high priority.
-        Priority priority = Priority.NORMAL;
-        if (category.charAt(0) == 'E' || category.charAt(0) == 'F' || category.charAt(0) == 'W') {
-            priority = Priority.HIGH;
+        String message = matcher.group(5);
+        String category = classifyIfEmpty(matcher.group(4), message);
+        Priority priority = null;
+        if (matcher.group(3) != null) {
+            try {
+                priority = Priority.fromString(matcher.group(3));
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/PyLintParser.java]: Can you please add a test case that contains such a priority.

File: src/main/java/hudson/plugins/warnings/parser/PyLintParser.java [hunk 2]
```diff
@@ -34,12 +43,22 @@ protected boolean isLineInteresting(final String line) {
...
+            } catch (IllegalArgumentException e) {
+                // will fall back to E/F/W = high
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/PyLintParser.java]: Can you please add a test case that contains an invalid text as priority.

**Comment by uhafner** [comment]
We now have three pull request:
- https://github.com/jenkinsci/warnings-plugin/pull/57
- https://github.com/jenkinsci/warnings-plugin/pull/61
- https://github.com/jenkinsci/warnings-plugin/pull/62
@ianhinder, @astraw38, @AngledLuffa: should I now merge https://github.com/jenkinsci/warnings-plugin/pull/62 and then merge an adapted version of https://github.com/jenkinsci/warnings-plugin/pull/61?

**Comment by AngledLuffa** [comment]
Merging #62 into my change works for me.  I was thinking it would be best
to wait for a response from the PyLint people before merging #61, though,
as my changes here depend on the change there.

File: src/main/java/hudson/plugins/warnings/parser/PyLintParser.java (hunk 2 above)
Comment (AngledLuffa) [inline src/main/java/hudson/plugins/warnings/parser/PyLintParser.java]: Of course.  I just added two test cases.  Let me know if you think this is
sufficient.

**Comment by uhafner** [comment]
Ping.

**Comment by astraw38** [comment]
If the priority is optional, I see no issue with merging it (62, then 61). We normally don't use pylint options that include any priority other than just the Failure/Error/Warning/Convention.

**Comment by uhafner** [comment]
Ok, I merged #62 now, which is already part of the latest release.

If you still need this one, please update to the latest commit and reopen.

**Pull request closed without merging**